cli/src/utilities/publishToGitHub.ts

```diff
@@ -0,0 +1,162 @@
+import { GluegunToolbox } from 'gluegun';
+import { execSync } from 'child_process';
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+const verifyRepoDeleted = (repoName: string): boolean => {
+  try {
+    execSync(`gh repo view ${repoName}`, { stdio: 'ignore' });
+    return false; // Repository still exists
+  } catch (e) {
+    return true; // Repository is deleted
+  }
+};
+
+export const publishToGitHub = async (toolbox: GluegunToolbox, projectName: string): Promise<string | null> => {
+  const {
+    print: { info, error }
+  } = toolbox;
+
+  try {
+    // Check if gh CLI is installed
+    try {
+      execSync('gh --version', { stdio: 'ignore' });
+    } catch (e) {
+      error('\nGitHub CLI (gh) is not installed.');
+      info('Please install it from: https://cli.github.com/');
+      info('Then run: gh auth login');
+      return null;
+    }
+
+    // Check if user is authenticated with correct permissions
+    try {
+      const scopes = execSync('gh auth status -t 2>&1', { encoding: 'utf8' });
+      if (!scopes.includes('delete_repo')) {
+        error('\nMissing required GitHub permissions.');
+        info('Please run the following command to grant delete repository permissions:');
+        info('  gh auth refresh -h github.com -s delete_repo');
+        return null;
+      }
+    } catch (e) {
+      error('\nYou are not authenticated with GitHub CLI.');
+      info('Please run: gh auth login');
+      return null;
+    }
+
+    // Check if repository already exists
+    try {
+      execSync(`gh repo view ${projectName}`, { stdio: 'ignore' });
+
+      // Repository exists, ask user if they want to delete it
+      const response = await toolbox.prompt.ask({
+        type: 'confirm',
+        name: 'deleteRepo',
+        message: `Repository "${projectName}" already exists and has content. Would you like to delete it?`
+      });
+
+      if (!response.deleteRepo) {
+        info('\nPublication cancelled.');
+        return null;
+      }
+
+      // Delete the existing repository
+      info('\nDeleting existing repository...');
+      try {
+        execSync(`gh repo delete ${projectName} --yes`, { stdio: 'ignore' });
```

Comment:
I was nervous about passing in an empty `projectName`, but it seems the GitHub CLI will determine the project name from the git repository, so the `projectName` argument could optionally be omitted.

I prefer _not_ omitting the `projectName`, that way we are sure exactly what prints to the user in the prompt and what we passed to `gh repo view` is used in the same way.

So no changes here :+1: